pyhanko/sign/signers/csc_signer.py

```diff
@@ -45,6 +45,55 @@ class TPrepDocumentResponse:
     pdf: TPreparedPdf
     digest: TDigest
 
+    class Asn1(core.Sequence):
```

Comment:
Hmmm. I'm certainly not averse to ASN.1-based workflows in general, but I don't think this particular situation warrants ASN.1 serialisation, especially since basically everything is an `OctetString` now, so we're not really getting all that much mileage out of the ASN.1 encoding anyway. Well, I suppose that tagging the lengths of each of the components is a good thing, but I'm not convinced yet.

To do this properly with `asn1crypto`, we'd have to define some new constructs to serialise `PostSignInstructions` and `PreparedByterangeDigest` the right way. There are three reasons why I'm not sold on doing that just yet:

 - Getting the type definitions right will be tricky, particularly for `PostSignInstructions`.
 - I'm  not sure how well `asn1crypto` deals with large unstructured payloads (which a general PDF file would certainly be), so performance may be a concern. 
 - If we decide to go this route, the ASN.1 type definitions need to go into another, more general module (but that's easy to solve).


So, I see two ways forward:

 - We go through the effort of defining ASN.1 types for all the things we need, and put them in a separate module. I'd serialise the PDF document content separately either way, because that's less likely to lead to performance issues.
 - We use a more JSON-like object model that also has binary-friendly features, like CBOR. That would add a dependency, but we can make that optional.

Independently of this, I think it'd be good to define a generic serialisation API that people can slot into, because probably not everyone wants/needs to do it the same way.

Thoughts?